lib/chain/src/ChainStore.js

```diff
@@ -1667,6 +1666,103 @@ class ChainStore {
         });
         return didImpact;
     }
+
+    getLiquidityPoolsByAssets(assetType = 'asset_a', assetA, assetB, limit = 101, startId) {
+        const ASSET_TYPE = [
+            'asset_a',
+            'asset_b',
+            'both',
+        ];
+        if (ASSET_TYPE.indexOf(assetType) === -1) {
+            throw Error("asset type is unexpected");
+        }
+        const METHOD_NAME = {
+            asset_a: 'get_liquidity_pools_by_asset_a',
+            asset_b: 'get_liquidity_pools_by_asset_b',
+            both: 'get_liquidity_pools_by_both_assets',
```

Comment:
Just FYI we have a new API `get_liquidity_pools_by_one_asset` which is added in `5.2.0` and is easier to use. See https://github.com/bitshares/bitshares-core/pull/2365 .